cmd/posix.go

```diff
@@ -98,13 +99,30 @@ func isDirEmpty(dirname string) bool {
 }
 
 // Initialize a new storage disk.
-func newPosix(diskPath string) (StorageAPI, error) {
-	if diskPath == "" {
+func newPosix(ep *url.URL) (StorageAPI, error) {
+	if ep == nil {
 		return nil, errInvalidArgument
 	}
-	suppliedDiskPath := diskPath
-	var err error
+	suppliedDiskPath := ep.String()
+	var diskPath string
+	// For windows ep.Path is usually empty
+	if runtime.GOOS == "windows" {
+		// For full URLs windows drive is part of URL path.
+		// Eg: http://ip:port/C:\mydrive
+		if ep.Scheme == "http" || ep.Scheme == "https" {
```

Comment:
at the posix layer, dealing with URL related fields does not look right.
If it is needed for printing, printing should be done differently. Because why should posix's String() print http/ipaddress related info?

generally if a function takes arguments that only it is concerned with it will result in a simpler code with right abstraction.